src/main/java/games/strategy/engine/lobby/client/login/LoginPanel.java

```diff
@@ -109,55 +97,35 @@ private void setupListeners() {
       dialog.setVisible(false);
     });
     cancel.addActionListener(e -> dialog.setVisible(false));
-    anonymous.addActionListener(e -> setWidgetActivation());
+    anonymousLogin.addActionListener(e -> updateComponents());
 
     SwingComponents.addEnterKeyListener(this, this::logonPressed);
   }
 
   private void logonPressed() {
-    final String userName = this.userName.getText();
-    final boolean anonymous = this.anonymous.isSelected();
-    if (!DBUser.isValidUserName(userName)) {
-      JOptionPane.showMessageDialog(this, DBUser.getUserNameValidationErrorMessage(userName), "Invalid Username",
-          JOptionPane.ERROR_MESSAGE);
+    if (!DBUser.isValidUserName(userName.getText())) {
+      JOptionPane.showMessageDialog(this, DBUser.getUserNameValidationErrorMessage(userName.getText()),
+          "Invalid Username", JOptionPane.ERROR_MESSAGE);
       return;
-    } else if (password.getPassword().length == 0 && !anonymous) {
-      JOptionPane.showMessageDialog(LoginPanel.this, "You must enter a password", "No Password",
-          JOptionPane.ERROR_MESSAGE);
+    } else if (password.getPassword().length == 0 && !anonymousLogin.isSelected()) {
+      JOptionPane.showMessageDialog(this, "You must enter a password", "No Password", JOptionPane.ERROR_MESSAGE);
       return;
-    } else if (password.getPassword().length < 3 && !anonymous) {
-      JOptionPane.showMessageDialog(LoginPanel.this, "Passwords must be at least three characters long",
-          "Invalid password", JOptionPane.ERROR_MESSAGE);
+    } else if (password.getPassword().length < 3 && !anonymousLogin.isSelected()) {
+      JOptionPane.showMessageDialog(this, "Passwords must be at least three characters long", "Invalid Password",
+          JOptionPane.ERROR_MESSAGE);
       return;
     }
+
     returnValue = ReturnValue.LOGON;
     dialog.setVisible(false);
   }
 
-  static void storePrefs(final String userName, final boolean anonymous) {
-    final Preferences prefs = Preferences.userNodeForPackage(LoginPanel.class);
-    prefs.put(LAST_LOGIN_NAME_PREF, userName);
-    prefs.putBoolean(ANONYMOUS_LOGIN_PREF, anonymous);
-    try {
-      prefs.flush();
-    } catch (final BackingStoreException e) {
```

Comment:
Explicitly flushing a preference node is not required (i.e. you won't find equivalent code in `LobbyLoginPreferences`).

Per the Javadocs, the JVM is guaranteed to flush all pending preference updates before it is shut down.  Since we don't do anything special to recover from the case where the backing store fails, there's no point in explicitly flushing the preference node (other than it allows someone to look at the preference store outside of the JVM while it is running and see the updates).